homeassistant/components/zabbix/__init__.py

```diff
@@ -46,13 +73,188 @@ def setup(hass, config):
+            return
+
+        entity_id = state.entity_id
+        if not entities_filter(entity_id):
+            return
+
+        floats = {}
+        strings = {}
+        try:
+            _state_as_value = float(state.state)
+            floats[entity_id] = _state_as_value
+        except ValueError:
+            try:
+                _state_as_value = float(state_helper.state_as_number(state))
+                floats[entity_id] = _state_as_value
+            except ValueError:
+                strings[entity_id] = state.state
+
+        for key, value in state.attributes.items():
+            # For each value we try to cast it as float
+            # But if we can not do it we store the value
+            # as string
+            attribute_id = f"{entity_id}/{key}"
+            try:
+                float_value = float(value)
+            except (ValueError, TypeError):
+                float_value = None
+            if float_value is None or not math.isfinite(float_value):
+                strings[attribute_id] = str(value)
+            else:
+                floats[attribute_id] = float_value
+
+        metrics = []
+        float_keys_count = len(float_keys)
+        float_keys.update(floats)
+        if len(float_keys) != float_keys_count:
+            floats_discovery = []
+            for float_key in float_keys:
+                floats_discovery.append({"{#KEY}": float_key})
+            metric = ZabbixMetric(
+                publish_states_host,
+                "homeassistant.floats_discovery",
+                json.dumps(floats_discovery),
+            )
+            metrics.append(metric)
+        for key, value in floats.items():
+            metric = ZabbixMetric(
+                publish_states_host, f"homeassistant.float[{key}]", value
+            )
+            metrics.append(metric)
+
+        string_keys.update(strings.keys())
+        return metrics
+
+    if publish_states_host:
+        zabbix_sender = ZabbixSender(zabbix_server=conf[CONF_HOST])
+        instance = ZabbixThread(hass, zabbix_sender, event_to_metrics)
+        instance.setup(hass)
+
     return True
+
+
+class ZabbixThread(threading.Thread):
+    """A threaded event handler class."""
+
+    MAX_TRIES = 3
+
+    def __init__(self, hass, zabbix_sender, event_to_metrics):
+        """Initialize the listener."""
+        threading.Thread.__init__(self, name="Zabbix")
+        self.queue = queue.Queue()
+        self.zabbix_sender = zabbix_sender
+        self.event_to_metrics = event_to_metrics
+        self.write_errors = 0
+        self.shutdown = False
+        self.float_keys = set()
+        self.string_keys = set()
+
+    def setup(self, hass):
+        """Set up the thread and start it."""
+        hass.bus.listen(EVENT_STATE_CHANGED, self._event_listener)
+        hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP, self._shutdown)
+        self.start()
+        _LOGGER.info("Started publishing state changes to Zabbix")
```

Comment:
```suggestion
        _LOGGER.debug("Started publishing state changes to Zabbix")
```

Reply:
Yes, but `info` also makes sense so the user knows that the optional configuration `publish_states_host` is used successfully. Anyway. changed to `debug`.